compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp

```diff
@@ -614,6 +615,104 @@ INTERCEPTOR(int, shutdown, int socket, int how) {
   return REAL(shutdown)(socket, how);
 }
 
+// I/O Multiplexing
+
+INTERCEPTOR(int, poll, struct pollfd *fds, nfds_t nfds, int timeout) {
+  __rtsan_notify_intercepted_call("poll");
+  return REAL(poll)(fds, nfds, timeout);
+}
+
+#if !SANITIZER_APPLE
+// FIXME: This should work on all unix systems, even Mac, but currently
+// it is showing some weird error while linking
+// error: declaration of 'select' has a different language linkage
```

Comment:
Drawing folks attention here, this is very strange and I haven't seen it for an interceptor before

Reply:
@cjappl  There might be some clues comparing the declarations of `select` on Darwin and other unix OSes. Did you already take a look? I can investigate when I have some time (probably not for a couple of weeks, though, unfortunately)

Reply:
I took a look but was not able to find anything interesting. Definitely let me know if you see something I didn't see, I'd be curious!